### PR TITLE
Dala 1014 manual maintenance sprint 33

### PR DESCRIPTION
**Pull request by SiegfriedSobkowiak**
# Manual Maintenance Sprint 33
# Maintenance tasks (to be completed by the assignee)
### Skipped updates
The following known issues need to be reviewed in case a compatible version is available. Add new known issues as they appear.
- [x] ktlint 0.45.2 (higher version is not compatible with jlleitschuh plugin)
- [x] primevue 3.15.0 (higher versions have issues with AutoComplete PrimeVue issue 2881)
- [x] primeicon 5.0.0 (higher version require higher version of primevue)
- [x] logback-classic and logback-core 1.2.11 (higher version break backend)
- [x] slf4j-api 1.7.36 (1.7.x is required in current setup otherwise logging breaks)
- [x] io.gitlab.arturbosch.detekt:detekt-cli 1.21.0 (Failed to compile)
- [x] sonarqube 3.4.0.2513 not update to 3.5.X, due to issues in file resolving mechanism
- [x] springboot 2.7.7 not updated to 3.0.X, due to issues when resolving main class name in api key manager 

### Gradle update
- [x] Execute `gradlew dependencyUpdates` to get a report on Dependencies with updates
- [x] Update `settings.gradle.kts` (for libraries), `build.gradle.kts` (for plugins) and `gradle.properties` (for jacoco and ktlint)
  Note: fasterXML is managed by spring, thus NO manual version update should be conducted
- [x] update the gradle wrapper: execute `gradle wrapper --gradle-version X.Y.Z`

### Dataland frontend
- [x] Update node version in `dataland-frontend/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in package.json
- [x] Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check for security issues

### Dataland keycloak theme
- [x] Update node version in `dataland-keycloak/dataland_theme/login/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in package.json
- [x]   Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check for security issues

### Dockerfile updates
Update versions in the following dockerfiles
- [x] `./dataland-api-key-manager/Dockerfile`
- [x] `./dataland-api-key-manager/DockerfileTest`
- [x] `./dataland-backend/DockerfileBase`
- [x] `./dataland-backend/DockerfileTest`
- [x] `./dataland-backend/Dockerfile`
- [x] `./dataland-csvconverter/Dockerfile`
- [x] `./dataland-e2etests/DockerfileBase`
- [x] `./dataland-frontend/Dockerfile`
- [x] `./dataland-frontend/DockerfileTest`
- [x] `./dataland-inbound-admin-proxy/Dockerfile`
- [x] `./dataland-inbound-proxy/DockerfileBase`
- [x] `./dataland-inbound-proxy/Dockerfile`
- [x] `./dataland-keycloak/Dockerfile`  (also update realm json files with new version)
- [x] `./base-dockerfiles/DockerfileGradle`
- [x] Update the versions of the external images for api-key-manager-db, backend-db, keycloak-db, internal-storage-db and frontend-dev
- [x] Check if there are any services in the `docker-compose.yml` file that have not gotten an update yet (e.g. a new service that is not covered by the tasks above)

## Server updates
Execute `sudo apt-get update && sudo apt-get upgrade` on
- [x] dev1.dataland.com
- [x] dev2.dataland.com
- [x] test.dataland.com
- [x] tunnel.dataland.com
- [x] letsencrypt.dataland.com
- [x] (OPT) dataland.com

## Check e-mails sent by keycloak
- [x] Check that the verification e-mail is sent and displayed correctly
- [x] Check that the reset password e-mail is sent and displayed correctly
- [x] Check that the password can be reset

## Conclusion
- [x] After updating all components check if everything still works
- [x] The new version is deployed to the dev server using this branch and real data
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that real data has been used
  - [x] It's verified that everything seems to be working fine by manually using the website
- [x] This template has been updated to reflect the latest state of tasks required and known issues with upgrades

# Review (to be completed by the reviewer)
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github.
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described above
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server using this branch
  - [x] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
  - [x] It's verified that everything seems to be working fine by manually using the website
  - [x] All implemented Social Logins have been tested manually in the UI
  - [x] Go to the swagger-UI, authorize, run a "GET" request to the companies endpoint and assure that your authorization has worked by assuring that you get a 200 response

- [x] If any work on the UI is to be merged, those changes were also documented in the Figma
- [x] The local Dev stack still works: execute `startDevelopmentStack.sh` and execute `npm run testpipeline -- --env EXECUTION_ENVIRONMENT=""` in dataland-frontend
- [x] After(!) the cypress tests have passed locally, execute the backend-e2e-tests `./gradlew dataland-e2etests:test`
- [x] Locally: Go to the swagger-UI, authorize, run a "GET" request to the companies endpoint and assure that your authorization has worked by assuring that you get a 200 response

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -46,25 +46,25 @@ tasks.dependencyUpdates.configure {
...
     detekt("io.gitlab.arturbosch.detekt:detekt-cli:1.20.0")
-    detekt("org.jetbrains.kotlin:kotlin-compiler-embeddable:1.7.21")
+    detekt("org.jetbrains.kotlin:kotlin-compiler-embeddable:1.8.0")
 }
 
 java.sourceCompatibility = JavaVersion.VERSION_17
 
 plugins {
     id("io.gitlab.arturbosch.detekt") version "1.20.0"
     id("com.github.node-gradle.node") version "3.5.0" apply false
```
Comment (ppuetsch) [inline build.gradle.kts:56]: com.github.node-gradle.node:com.github.node-gradle.node.gradle.plugin [3.5.0 -> 3.5.1]

File: build.gradle.kts [hunk 2]
```diff
@@ -46,25 +46,25 @@ tasks.dependencyUpdates.configure {
...
-    kotlin("plugin.spring") version "1.7.21" apply false
+    kotlin("jvm") version "1.8.0"
+    kotlin("plugin.spring") version "1.8.0" apply false
     id("org.sonarqube") version "3.4.0.2513"
     jacoco
-    id("org.springdoc.openapi-gradle-plugin") version "1.4.0" apply false
+    id("org.springdoc.openapi-gradle-plugin") version "1.6.0" apply false
     id("com.gorylenko.gradle-git-properties") version "2.4.1" apply false
     id("org.openapi.generator") version "6.2.1" apply false
     id("com.github.ben-manes.versions") version "0.43.0"
```
Comment (ppuetsch) [inline build.gradle.kts:66]: - com.github.ben-manes.versions:com.github.ben-manes.versions.gradle.plugin [0.43.0 -> 0.44.0]

File: build.gradle.kts [hunk 3]
```diff
@@ -46,25 +46,25 @@ tasks.dependencyUpdates.configure {
 
 dependencies {
     detekt("io.gitlab.arturbosch.detekt:detekt-cli:1.20.0")
```
Comment (ppuetsch) [inline build.gradle.kts:48]: io.gitlab.arturbosch.detekt:io.gitlab.arturbosch.detekt.gradle.plugin [1.20.0 -> 1.22.0]

1.21.0 had known issues - what about 1.22.?

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md [hunk 4]
```diff
@@ -21,7 +23,7 @@ The following known issues need to be reviewed in case a compatible version is a
 ### Dataland frontend
```
Comment (ppuetsch) [inline .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:23]: I Think
```
 Note: fasterXML is managed by spring, thus NO manual version update should be conducted
```
can be removed.

File: dataland-frontend/package.json [hunk 5]
```diff
@@ -39,28 +39,28 @@
...
+    "npm-check-updates": "^16.6.2",
     "otplib": "^12.0.1",
     "ts-node": "^10.9.1",
-    "tsconfig-paths": "^4.1.0",
-    "typescript": "^4.8.4",
-    "vite": "^3.2.3",
-    "vite-plugin-istanbul": "^3.0.2"
+    "tsconfig-paths": "^4.1.2",
+    "typescript": "^4.9.4",
+    "vite": "^4.0.3",
```
Comment (ppuetsch) [inline dataland-frontend/package.json:62]: vite 4.0.4

File: dataland-frontend/package.json [hunk 6]
```diff
@@ -39,28 +39,28 @@
...
     "@types/big.js": "^6.1.6",
     "@types/json2csv": "^5.0.3",
-    "@types/node": "^18.11.9",
-    "@vitejs/plugin-vue": "^3.2.0",
+    "@types/node": "^18.11.18",
+    "@vitejs/plugin-vue": "^4.0.0",
     "@vue/eslint-config-prettier": "^7.0.0",
     "@vue/eslint-config-typescript": "^11.0.2",
-    "@vue/test-utils": "^2.2.1",
+    "@vue/test-utils": "^2.2.6",
```
Comment (ppuetsch) [inline dataland-frontend/package.json:46]: 2.2.7

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -14,5 +14,3 @@ Please check all boxes before the Pull Request is merged. In case you skip a box
   - [ ] The new feature is manually used/tested/observed on dev server
```
Comment (ppuetsch) [inline .github/pull_request_template.md:14]: kannst du hier auch noch gitinfo (statt actuator/info) reinschreiben?

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md (hunk 4 above)
Comment (SiegfriedSobkowiak) [reply .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:23]: Yes it did not show up lately.

File: build.gradle.kts (hunk 1 above)
Comment (SiegfriedSobkowiak) [reply build.gradle.kts:56]: updated

File: build.gradle.kts (hunk 2 above)
Comment (SiegfriedSobkowiak) [reply build.gradle.kts:66]: updated

File: dataland-frontend/package.json (hunk 5 above)
Comment (SiegfriedSobkowiak) [reply dataland-frontend/package.json:62]: updated

File: dataland-frontend/package.json (hunk 6 above)
Comment (SiegfriedSobkowiak) [reply dataland-frontend/package.json:46]: updated

File: .github/pull_request_template.md (hunk 7 above)
Comment (SiegfriedSobkowiak) [reply .github/pull_request_template.md:14]: erledigt

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md [hunk 8]
```diff
@@ -66,7 +68,7 @@ Execute `sudo apt-get update && sudo apt-get upgrade` on
 ## Conclusion
 - [ ] After updating all components check if everything still works
 - [ ] The new version is deployed to the dev server using this branch and real data
-  - [ ] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
+  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
   - [ ] It's verified that real data has been used
   - [ ] It's verified that everything seems to be working fine by manually using the website
 - [ ] This template has been updated to reflect the latest state of tasks required and known issues with upgrades
```
[the remark this answers is not in the document]
Comment (SiegfriedSobkowiak) [reply .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:74]: erledigt

File: build.gradle.kts (hunk 1 above)
Comment (ppuetsch) [reply build.gradle.kts]: ok

File: build.gradle.kts (hunk 3 above)
Comment (ppuetsch) [reply build.gradle.kts:48]: ok

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md (hunk 4 above)
Comment (ppuetsch) [reply .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:22]: ok

File: .github/pull_request_template.md (hunk 7 above)
Comment (ppuetsch) [reply .github/pull_request_template.md:14]: ok

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ppuetsch) [reply .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:73]: ok

File: build.gradle.kts (hunk 2 above)
Comment (ppuetsch) [reply build.gradle.kts]: ok

File: dataland-frontend/package.json (hunk 5 above)
Comment (ppuetsch) [reply dataland-frontend/package.json]: ok

File: dataland-frontend/package.json (hunk 6 above)
Comment (ppuetsch) [reply dataland-frontend/package.json]: ok

**Pull request merged**